detox/src/DetoxWorker.js

```diff
@@ -258,8 +258,8 @@ class DetoxWorker {
       testName: testSummary.fullName,
     });
 
-    if (copilot.isInitialized()) {
-      copilot.end(testSummary.status !== 'passed');
+    if (this.pilot.isInitialized()) {
```

Comment:
Here, `isInitialized` is part of `this.pilot` (i.e. `DetoxPilot` and not the core `Pilot`), and in the failing tests you're trying to mock a method that is called `isInitialized` of the core `Pilot`)